src/device.hpp

```diff
@@ -210,9 +209,12 @@ typedef struct _cl_device_id {
     uint32_t m_vulkan_queue_alloc_index;
 
     std::string m_extensions;
+};
 
-} cvk_device;
+static inline cvk_device* icd_downcast(cl_device_id device) {
+    return static_cast<cvk_device*>(device);
+}
 
-typedef struct _cl_platform_id {
+typedef struct _cl_platform_id : public clvk::icd_object {
```

Comment:
Any specific reason `cvk_platform` is declared differently from all other objects?